docs: document computeDrift() more clearly

analytically opened the pull request:
Improve component scheduling performance with two key optimizations:

1. Implement time-based caching for drift calculations with a 100ms TTL to reduce redundant calls to NumGoroutine() and subsequent calculations

2. Replace all floating-point operations with more efficient integer arithmetic while preserving identical behavior

These changes should significantly reduce CPU usage in busy Concourse deployments where IntervalElapsed() is called frequently. The pure integer implementation avoids type conversions and uses the more efficient CPU operations, while the caching strategy prevents recalculating drift values unnecessarily.

No interface changes or behavior modifications - this is a pure performance optimization that maintains full backward compatibility.

taylorsilva reviewed (commented):
I'm not sure this is useful sadly. The piece of info I think you're missing here is that `IntervalElapsed()` is only called every ~10s, which means the drift cache of 100ms is always invalid. We never end up using the cached drift.

I threw in some `fmt.Println()` statements in the `IntervalElapsed()` func to verify this:
```diff
func (c *component) IntervalElapsed() bool {
	now := c.clock.Now()
+       fmt.Println("taydev | component: ", c.name, "IntervalElapsed()", now)

	// Only recompute drift if enough time has passed or first call
	if now.Sub(c.lastDriftTime) > c.driftCacheTTL || c.lastDriftTime.IsZero() {
+               fmt.Println("taydev | component: ", c.name, "computeDrift()")
		c.lastDrift = c.computeDrift()
		c.lastDriftTime = now
	}

```

Ran `docker compose up -d --build` and then after a minute I checked the logs and saw this:
```
┌─[compinterval*][taylor-macbook][~/workspace/concourse/concourse]
└─▪ docker compose logs web | grep taydev | grep collector_pipelines
web-1  | taydev | component:  collector_pipelines IntervalElapsed() 2025-03-17 16:34:09.049589964 +0000 UTC m=+10.273548047
web-1  | taydev | component:  collector_pipelines computeDrift()
web-1  | taydev | component:  collector_pipelines IntervalElapsed() 2025-03-17 16:34:19.063078177 +0000 UTC m=+20.287036261
web-1  | taydev | component:  collector_pipelines computeDrift()
web-1  | taydev | component:  collector_pipelines IntervalElapsed() 2025-03-17 16:34:29.065698584 +0000 UTC m=+30.290623098
web-1  | taydev | component:  collector_pipelines computeDrift()
web-1  | taydev | component:  collector_pipelines IntervalElapsed() 2025-03-17 16:34:39.070149172 +0000 UTC m=+40.295073727
web-1  | taydev | component:  collector_pipelines computeDrift()
web-1  | taydev | component:  collector_pipelines IntervalElapsed() 2025-03-17 16:34:49.087007719 +0000 UTC m=+50.311932274
web-1  | taydev | component:  collector_pipelines computeDrift()
web-1  | taydev | component:  collector_pipelines IntervalElapsed() 2025-03-17 16:34:59.090145959 +0000 UTC m=+60.316433612
web-1  | taydev | component:  collector_pipelines computeDrift()
web-1  | taydev | component:  collector_pipelines IntervalElapsed() 2025-03-17 16:35:09.099098505 +0000 UTC m=+70.325386158
web-1  | taydev | component:  collector_pipelines computeDrift()
```

I do like the expanded comment block you added here though. Makes it MUCH clearer how `computeDrift()` works. Would be happy to take that piece in.

analytically commented:
Fair enough. I mistakingly was under the impression that these component updates were called much more often.

analytically committed:
docs: document computeDrift() more clearly

No interface changes or behavior modifications - this is a small
readability optimization that maintains full backward compatibility.

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>

taylorsilva committed:
go modernize: use min() to return the drift

Signed-off-by: Taylor Silva <dev@taydev.net>

taylorsilva reviewed (approved):
LGTM - Concourse ships as one binary, but its code operates as a distributed system. There are a lot of moving pieces and understanding how they all work together takes some time :)
